src/mds/Server.cc

```diff
@@ -5722,6 +5722,34 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string quota_str)
+{
+  int64_t quota_str_len = quota_str.length();
+  if (quota_str_len < 1) {
+    return -CEPHFS_EINVAL;
+  }
+  if (quota_str_len == 1) {
+    if (isdigit(quota_str[0])) { return stoi(quota_str); } 
+    else { return -CEPHFS_EINVAL; }
+  }
+  map<char, int64_t> size_value = {{'k', 1024}, {'m', pow(1024, 2)}, 
+                                   {'g', pow(1024, 3)}, {'t', pow(1024, 4)}};
+  for(char quota_digit: quota_str.substr(0, quota_str_len - 1)) {
+    if(!isdigit(quota_digit)) {
+      return -CEPHFS_EINVAL;
+    }
+  }
+  char last_ch = tolower(quota_str[quota_str_len - 1]);
+  if (isdigit(last_ch)) {
+    return stoi(quota_str);
+  }
+  if (size_value.find(last_ch) != size_value.end()) {
+    return stoi(quota_str.substr(0, quota_str_len - 1)) * size_value[last_ch];
+  }
+  derr << "not supp" << dendl;
+  return -CEPHFS_EINVAL;
+}
+
```

Comment:
this could be made simpler like ...
```
auto last_char = quota_str.back();
if (size_value.find(last_char) != size_value.end()) {
  try {
    return boost::lexical_cast<int64_t>(quota_str.substr(0, quota_str.length()-1)) * size_value[last_char);
  } catch (boost:bad_lexical_cast&) {
    return -EINVAL;
  }
} else {
  try {
    return boost::lexical_cast<int64_t>(quota_str);
  } catch (boost:bad_lexical_cast&) {
    return -EINVAL;
  }
}
```

- You should let `boost::lexical_cast<int64_t>()`  do the heavy lifting for you. That's exactly what its designed for.
- You should use `std::unordered_map` instead of `std::map` in this case
- You could avoid dependency on the math library by specifying bit shifts for the multiplier values
- You could pass the `value` as a string reference for the call to `return_bytes()`

Reply:
Ah, okay, will try dig into boost now. thanks for the suggestions. will work on it and push changes asap.

Reply:
done